Homeworks/04/src/components/Fetch.tsx

```diff
@@ -10,26 +10,43 @@ interface PokemonResponse {
 
 export default function Fetch() {
   const [pokemon, setPokemon] = useState<PokemonResponse[]>([])
+  const [id, setId] = useState<number>(1)
 
   // your task here is to fetch Pokémon from a given URL and just display its name
   // first Pokémon should be fetched when the compontent is first rendered
   // then, once user clicks on FETCH button, increment your id to fetch the next one and add it to the list
 
+  useEffect(() => {
+    fetchPokemon(id).then(res => {
+      setPokemon(pokemon => [...pokemon, res])
+    })
+  }, [])
```

Comment:
Iz React [dokumentacije](https://react.dev/reference/react/useEffect):

> useEffect is a React Hook that lets you [synchronize a component with an external system.](https://react.dev/learn/synchronizing-with-effects)

U ovom slučaju, želimo sinkronizirati našu lokalnu listu pokemona sa vanjskim API-em svaki puta kada se promjeni trenutni `id`. Zašto onda `id` nije dodan u dependency listu? Čak nam je i eslint dao upozorenje ovdje:

![Screenshot 2025-03-30 at 14 32 23](https://github.com/user-attachments/assets/14ae6445-f4da-4a4a-9a33-5728eaba0c39)

Kada bi dodali id u dependency array, to bi nam omogucilo da se novi podaci dohvate svaki puta kada se promjeni `id`, imali bi jasniji flow i ne bi morali duplicirati logiku kao što si ti dolje u `onClick` implementaciji.

```suggestion
  useEffect(() => {
    fetchPokemon(id).then(res => {
      setPokemon(pokemon => [...pokemon, res])
    })
  }, [id])
```

Logiku za sort također možemo ubaciti ovdje tako da se lista uvijek postavlja sortirana.